### ISSUE TITLE
🐛 False positive in noSelfAssign with properties on `this`

### ISSUE THREAD

**Issue by ZachHaber** [biomejs/biome#576]
### Environment information

```block
biome - playground
```

### What happened?

If you set a literal computed expression on a property on `this` equal to the same computed expression on any object name, it considers it a self assignment, and same in reverse.

```js
this.arr[0] = p[0];
p[0] = this.arr[0]
```

Reproduction with several test cases: [Playground](https://biomejs.dev/playground/?code=LwAvACAAYAB0AGgAaQBzAC4AYQBuAHkAdABoAGkAbgBnAFsAYQBuAHkAdABoAGkAbgBnAF0AYAAgAGkAcwAgAHQAaABlACAAcwBhAG0AZQAgAGEAcwAgAGAAYQBuAHkAbwB0AGgAZQByAHQAaABpAG4AZwBbAGEAbgB5AHQAaABpAG4AZwBdAGAACgB0AGgAaQBzAC4AYQByAHIAWwAwAF0AIAA9ACAAcABbADAAXQA7AAoALwAvACAAaQBuAHYAZQByAHQAZQBkAAoAcABbADAAXQAgAD0AIAB0AGgAaQBzAC4AYQByAHIAWwAwAF0AOwAKAC8ALwAgAGEAbgB5ACAAbwBiAGoAZQBjAHQAIABuAGEAbQBlAAoAdABoAGkAcwAuAGEAcgByAFsAMABdACAAPQAgAGEAcgByADIAWwAwAF0ACgAvAC8AIABvAHQAaABlAHIAIABjAGwAYQBzAHMAIABmAGkAZQBsAGQAcwAKAHQAaABpAHMALgBhAHIAcgBbADAAXQAgAD0AIAB0AGgAaQBzAC4AYQByAHIAMgBbADAAXQAKAC8ALwAgAGIAaQBvAG0AZQAtAGkAZwBuAG8AcgBlACAAbABpAG4AdAAvAGMAbwBtAHAAbABlAHgAaQB0AHkALwB1AHMAZQBMAGkAdABlAHIAYQBsAEsAZQB5AHMAOgAgAGkAcgByAGUAbABlAHYAYQBuAHQACgB0AGgAaQBzAC4AYQByAHIAWwAnAGEAJwBdACAAPQAgAHAAWwAnAGEAJwBdADsACgAvAC8AIABvAG4AbAB5ACAAdwBpAHQAaAAgAGAAdABoAGkAcwBgAAoAYwBvAG4AcwB0ACAAYQByAHIAIAA9ACAAdABoAGkAcwAuAGEAcgByADsACgBhAHIAcgBbADAAXQAgAD0AIABwAFsAMABdAAoALwAvACAAbwBuAGwAeQAgAGMAbwBtAHAAdQB0AGUAZAAgAGUAeABwAHIAZQBzAHMAaQBvAG4AcwAgAAoAdABoAGkAcwAuAGEAcgByAC4AYQAgAD0AIABwAC4AYQA7AAoALwAvACAAbwBuAGwAeQAgAGwAaQB0AGUAcgBhAGwAIABjAG8AbQBwAHUAdABlAGQAIABlAHgAcAByAGUAcwBzAGkAbwBuAHMACgBjAG8AbgBzAHQAIABhACAAPQAgACcAYQAnADsACgB0AGgAaQBzAC4AYQByAHIAWwBhAF0AIAA9ACAAcABbAGEAXQA7AA%3D%3D)

### Expected result

It should not count things that aren't a self-assignment as a self assignment.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
